WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt

```diff
@@ -315,31 +315,35 @@ class WPMainNavigationView @JvmOverloads constructor(
             return fragment
         }
 
+
         internal fun getFragment(position: Int): Fragment? {
             return pages().getOrNull(position)?.let { pageType ->
-                val fragment = fragmentManager?.findFragmentByTag(getTagForPageType(pageType))
-                if (fragment != null) {
-                    when(fragment){
-                        is ReaderFragment, is NotificationsListFragment -> {
-                            return if (jetpackFeatureRemovalPhaseHelper.shouldShowStaticPage()) {
-                                fragmentManager?.beginTransaction()?.remove(fragment)?.commitNow()
-                                createFragment(pageType, jetpackFeatureRemovalPhaseHelper)
-                            } else {
-                                fragment
-                            }
-                        }
-                        is JetpackStaticPosterFragment -> {
-                            return if (!jetpackFeatureRemovalPhaseHelper.shouldShowStaticPage()) {
-                                fragmentManager?.beginTransaction()?.remove(fragment)?.commitNow()
-                                createFragment(pageType, jetpackFeatureRemovalPhaseHelper)
-                            } else {
-                                fragment
-                            }
-                        }
+                val currentFragment = fragmentManager?.findFragmentByTag(getTagForPageType(pageType))
+                return currentFragment?.let {
+                    when(it){
+                        is ReaderFragment, is NotificationsListFragment -> checkAndCreateForStaticPage(it, pageType)
+                        is JetpackStaticPosterFragment -> checkAndCreateForNonStaticPage(it, pageType)
+                        else -> { it}
```

Comment:
I really like how this code is evolving 👍🏻 
I'm wondering though if we really need the braces around `it`, doesn't it work without them? To me that reads cleaner

Reply:
We don't need them @ovitrif; it's our time constraints rearing it's ugly head - don't feel like I have time to go review properly, so thanks for catching it. 🙇

Reply:
If it helps I can take care of small little things like this, I really don't push or anything for making everything as good as I could envision it at the moment, it's also not the most critical thing so from my side I can live with it, if fixing these small details take from our more important focus 👍🏻

Reply:
❤️  I fixed this one in [65962a1](https://github.com/wordpress-mobile/WordPress-Android/pull/18120/commits/65962a136f638a1fade225635ca860450f10cd1b)

Reply:
Thank you! LGTM 🏆